cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1020,6 +1060,23 @@ func generateMountsModeRecover(mst *initramfsMountsState) error {
 
 		return machine, nil
 	}()
+	if tryingCurrentSystem {
+		// end of the line for a recovery system we are only trying out,
+		// this branch always ends with a reboot (or a panic)
+		outcome := boot.TryRecoverySystemOutcomeFailure
+		if err == nil && !machine.degraded() {
+			outcome = boot.TryRecoverySystemOutcomeSuccess
+		}
+		if outcome == boot.TryRecoverySystemOutcomeFailure {
```

Comment:
maybe I'm missing something but wouldn't an else be clearer than setting outcome and then checking it again?